builtin.c

```diff
@@ -63,6 +63,30 @@ static void S_finish_export_lexical(pTHX)
     LEAVE;
 }
 
+XS(XS_builtin_load);
+XS(XS_builtin_load)
+{
+    dXSARGS;
+
+    if (items != 1) {
+        croak_xs_usage(cv, "arg");
+    }
+
+    SV *module = TOPs;
+
+    if (!SvOK(module)) {
+        Perl_croak(aTHX_ "Undefined argument to builtin::load");
+    }
+
+    module = newSVsv(module);
+
+    if (!SvPOK(module))
+        SvPV_force_nolen(module);
+
+    Perl_load_module(aTHX_ PERL_LOADMOD_NOIMPORT, module, NULL, NULL, NULL);
+
+    XSRETURN(1);
```

Comment:
What value are you intending to return here?  Right now it is returning the module name.  (which confused me when I had an extra FETCH call.)

Reply:
I assumed that was deliberate, but yeah it is a bit confusing and a comment about it may help.

Reply:
Perhaps you wanted `XSRETURN_YES`?

Reply:
While I don't think the return value from `require` is particularly useful, I think I'd rather have this function return the same thing. It would be much easier to explain or document this function if it behaved the same as `require` aside from the input value.

Reply:
I don't know that we can easily get at the return value of Perl_load_module?

Returning 1 I think makes the most sense, rather than "Whatever the package returned"...

Reply:
> I don't know that we can easily get at the return value of Perl_load_module?
> 
> Returning 1 I think makes the most sense, rather than "Whatever the package returned"...

That's not what `XSRETURN(1)` means.

It means "the return value is the 1 most recent items pushed to the SV stack". I.e. whatever was at the top of stack at the time.

I'd suggest it ought to just return "true", because that's what `require` would return for any module that has the "require on me just returns true" flag set - see other RFC/discussion.

Reply:
Yeah I know the current behaviour is wrong. I'm talking about "What do we want" :)

Reply:
I would strongly prefer it to return the same value as `require`.

The RFC states:
> The effect of C<load>-ing a module is the same as C<require>-ing, down to
> the same error conditions when the module does not exist, does not compile,
> or does not evalute to a true value.

While this doesn't state anything about the return value, it implies that `load` should behave the same in every way except for its input.

Reply:
As it is runtime thing returning a module name could be useful to load a class and call a method (chain) on it.
`load($class)->new->foo->bar;`